frontend/lib/resolution/Resolver.cpp

```diff
@@ -2194,6 +2220,28 @@ static void maybeEmitWarningsForId(Resolver* rv, QualifiedType qt,
   }
 }
 
+QualifiedType Resolver::getSuperType(Context* context,
+                                     const QualifiedType& sub,
+                                     const Identifier* identForError) {
+  // Early return: if we don't know the child type, we can't figure out the
+  // parent type either.
+  if (sub.isUnknownKindOrType()) {
+    return QualifiedType();
+  }
+
+  if (auto classType = sub.type()->toClassType()) {
+    auto basicParentClass = classType->basicClassType()->parentClassType();
+    auto newClassType = ClassType::get(context,
+        basicParentClass,
+        classType->manager(),
+        classType->decorator());
```

Comment:
I think that technically the parent type should be ``borrowed``, rather than copying the manager/decorator of the actual class. This probably wouldn't be an issue inside a method, but could be confusing for something like ``myVar.super``.